src/Uno.UI/Microsoft/UI/Xaml/Controls/ColorPicker/ColorSpectrum.cs

```diff
@@ -0,0 +1,1810 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Numerics;
+using System.Threading.Tasks;
+using Uno.Disposables;
+using Uno.Extensions;
+using Uno.UI.Helpers.WinUI;
+using Windows.Devices.Input;
+using Windows.Foundation;
+using Windows.System;
+using Windows.System.Threading;
+using Windows.UI;
+using Windows.UI.Core;
+using Windows.UI.Input;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Automation.Peers;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Input;
+using Windows.UI.Xaml.Media;
+using Windows.UI.Xaml.Media.Imaging;
+using Windows.UI.Xaml.Shapes;
+
+namespace Microsoft.UI.Xaml.Controls.Primitives
+{
+	public partial class ColorSpectrum : Control
+	{
+		private bool m_updatingColor;
+		private bool m_updatingHsvColor;
+		private bool m_isPointerOver;
+		private bool m_isPointerPressed;
+		private bool m_shouldShowLargeSelection;
+		private List<Hsv> m_hsvValues = new List<Hsv>(); // Uno Doc: Set an instance unlike WinUI
+
+		// XAML elements
+		private Grid m_layoutRoot;
+		private Grid m_sizingGrid;
+
+		private Rectangle m_spectrumRectangle;
+		private Ellipse m_spectrumEllipse;
+		private Rectangle m_spectrumOverlayRectangle;
+		private Ellipse m_spectrumOverlayEllipse;
+
+		private FrameworkElement m_inputTarget;
+		private Panel m_selectionEllipsePanel;
+
+		private ToolTip m_colorNameToolTip;
+
+		private IAsyncAction m_createImageBitmapAction;
+
+		// On RS1 and before, we put the spectrum images in a bitmap,
+		// which we then give to an ImageBrush.
+		private WriteableBitmap m_hueRedBitmap;
+		private WriteableBitmap m_hueYellowBitmap;
+		private WriteableBitmap m_hueGreenBitmap;
+		private WriteableBitmap m_hueCyanBitmap;
+		private WriteableBitmap m_hueBlueBitmap;
+		private WriteableBitmap m_huePurpleBitmap;
+
+		private WriteableBitmap m_saturationMinimumBitmap;
+		private WriteableBitmap m_saturationMaximumBitmap;
+
+		private WriteableBitmap m_valueBitmap;
+
+		// On RS2 and later, we put the spectrum images in a loaded image surface,
+		// which we then put into a SpectrumBrush.
+		// Uno Doc: SpectrumBrush is disabled for now
+		//private LoadedImageSurface m_hueRedSurface;
```

Comment:
Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6121259)